### DISCUSSION TITLE
What is the difference between IEnumerable and IQueryable?

### DISCUSSION THREAD

**Discussion by smartcat5** [community/community#138670] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am a csharp beginner.
What is the difference between IEnumerable and IQueryable?

**Comment by masterwww1** [comment, marked as answer]
IEnumerable and IQueryable are both interfaces in C# that are used to iterate over collections, but they differ in how and when they execute queries.

1. INumberable is used for querying in-memory collections but IQueryable is used for querying data from external sources.
2. IEnumerable is execute immediately but IQueryable executed lazily.
